azurerm/internal/services/media/media_streaming_endpoint_resource.go

```diff
@@ -0,0 +1,591 @@
+			}
+			akamaiSignatureHeaderAuthenticationKeyList[index] = akamaiSignatureHeaderAuthenticationKey
+			accessControlResult.Akamai = &media.AkamaiAccessControl{
+				AkamaiSignatureHeaderAuthenticationKeyList: &akamaiSignatureHeaderAuthenticationKeyList,
+			}
+		}
+	}
+
+	return accessControlResult
+}
+
+func flattenAccessControl(accessControl *media.StreamingEndpointAccessControl) []interface{} {
+	if accessControl == nil {
+		return make([]interface{}, 0)
+	}
+
+	result := make(map[string]interface{})
+
+	if accessControl.IP != nil {
+		allows := make([]interface{}, len(*accessControl.IP.Allow))
+		for i, ipAllow := range *accessControl.IP.Allow {
+			allow := make(map[string]interface{})
+			if ipAllow.Name != nil {
+				allow["name"] = ipAllow.Name
+			}
+			if ipAllow.Address != nil {
+				allow["address"] = ipAllow.Address
+			}
+			if ipAllow.SubnetPrefixLength != nil {
+				allow["subnet_prefix_length"] = ipAllow.SubnetPrefixLength
+			}
+			if allow != nil {
+				allows[i] = allow
+			}
+		}
+
+		result["ip_allow"] = allows
+	}
+
+	if accessControl.Akamai != nil {
+		akamaiSignatureKeyList := make([]interface{}, len(*accessControl.Akamai.AkamaiSignatureHeaderAuthenticationKeyList))
+		for i, key := range *accessControl.Akamai.AkamaiSignatureHeaderAuthenticationKeyList {
+			akamaiSignatureHeaderKey := make(map[string]interface{})
+			if key.Base64Key != nil {
+				akamaiSignatureHeaderKey["base64_key"] = key.Base64Key
+			}
+			if key.Expiration != nil {
+				akamaiSignatureHeaderKey["expiration"] = key.Expiration.Format(time.RFC3339)
+			}
+			if key.Identifier != nil {
+				akamaiSignatureHeaderKey["identifier"] = key.Identifier
+			}
+			if akamaiSignatureKeyList != nil {
+				akamaiSignatureKeyList[i] = akamaiSignatureHeaderKey
+			}
+		}
+
+		result["akamai_signature_header_authentication_key"] = akamaiSignatureKeyList
+	}
+
+	return []interface{}{result}
+}
+
+func expandCrossSiteAccessPolicies(d *schema.ResourceData) *media.CrossSiteAccessPolicies {
+	crossSiteAccessPolicies := d.Get("cross_site_access_policies").([]interface{})
+	crossSiteAccessPolicy := crossSiteAccessPolicies[0].(map[string]interface{})
+	clientAccessPolicy := crossSiteAccessPolicy["client_access_policy"].(string)
+	crossDomainPolicy := crossSiteAccessPolicy["cross_domain_policy"].(string)
+	return &media.CrossSiteAccessPolicies{
+		ClientAccessPolicy: &clientAccessPolicy,
+		CrossDomainPolicy:  &crossDomainPolicy,
+	}
+}
+
+func flattenCrossSiteAccessPolicies(crossSiteAccessPolicies *media.CrossSiteAccessPolicies) []interface{} {
+	if crossSiteAccessPolicies == nil {
+		return make([]interface{}, 0)
+	}
+
+	result := make(map[string]interface{})
+
+	if crossSiteAccessPolicies.ClientAccessPolicy != nil {
+		result["client_access_policy"] = *crossSiteAccessPolicies.ClientAccessPolicy
+	}
+	if crossSiteAccessPolicies.CrossDomainPolicy != nil {
+		result["cross_domain_policy"] = *crossSiteAccessPolicies.CrossDomainPolicy
+	}
+
+	return []interface{}{result}
```

Comment:
to ensure these fields are always set, accounting for a bad response from the API - this wants to become:

```suggestion
	clientAccessPolicy := ""
	if input.ClientAccessPolicy != nil {
		clientAccessPolicy = *input.ClientAccessPolicy
	}
	
	crossDomainPolicy := ""
	if input.CrossDomainPolicy != nil {
		crossDomainPolicy = *input.CrossDomainPolicy
	}
	
	return []interface{}{
		map[string]interface{}{
			"client_access_policy": clientAccessPolicy,
			"cross_domain_policy": crossDomainPolicy,
		},
	}
```